arrow/src/main/java/org/apache/calcite/adapter/arrow/ArrowTable.java

```diff
@@ -120,7 +120,11 @@ public Enumerable<Object> query(DataContext root, ImmutableIntList fields,
       projector = null;
 
       final List<TreeNode> conditionNodes = new ArrayList<>(conditions.size());
+      String op = conditions.get(conditions.size() - 1);
```

Comment:
Can we add a complex condition that includes different supported operators？ like
`a = ? and (b = ? or (a is null) ) `

Reply:
I refactored this part of the code, and the current code has almost no impact on the functionality of the original code.

cc @mihaibudiu

Reply:
done